src/test/java/com/microsoft/sqlserver/jdbc/resultset/ResultSetTest.java

```diff
@@ -510,4 +511,89 @@ public void testResultSetMethods() throws SQLException {
             }
         }
     }
+
+    @Test
+    public void testMultipleResultSets() throws SQLException {
+        try (Statement stmt = connection.createStatement()) {
+            stmt.execute(
+                    "CREATE TABLE " + AbstractSQLGenerator.escapeIdentifier(tableName) + " (c1 int IDENTITY, c2 int)");
+            String SQL = "exec sp_help 'dbo." + TestUtils.escapeSingleQuotes(tableName) + "'";
+
+            boolean results = stmt.execute(SQL);
+            int rsCount = 0;
+
+            // Loop through the available result sets.
+            while (results) {
+                try (ResultSet rs = stmt.getResultSet()) {
+                    rsCount++;
+                    String firstColumnName = rs.getMetaData().getColumnName(1);
+                    switch (rsCount) {
```

Comment:
This `switch` statement seems to be redundant, `firstColumnValue` can be fetched using the column names instead in the second `switch statement`, which will also validate the column names at the same time (if column names do not match, exception will be thrown). 

```
            // Loop through the available result sets.
            while (results) {
                try (ResultSet rs = stmt.getResultSet()) {
                    rsCount++;
                    int i = 1;
                    while (rs.next()) {
                        String firstColumnValue = null;rs.getString(1);
                        switch (rsCount) {
                            case 1:
                                firstColumnValue = rs.getString("Name");
                                assert (firstColumnValue.equals(tableName));
                                break;
                            case 2:
                                firstColumnValue = rs.getString("Column_name");
                                assert (firstColumnValue.equalsIgnoreCase("c" + i++));
                                break;
                            case 3:
                                firstColumnValue = rs.getString("Identity");
                                assert (firstColumnValue.equalsIgnoreCase("c1"));
                                break;
                            case 4:
                                firstColumnValue = rs.getString("RowGuidCol");
                                assert (firstColumnValue.equalsIgnoreCase("No rowguidcol column defined."));
                                break;
                            case 5:
                                firstColumnValue = rs.getString("Data_located_on_filegroup");
                                assert (firstColumnValue.equalsIgnoreCase("PRIMARY"));
                                break;
                        }
                    }
                }
```